addons/mail/static/src/model/model_core.js

```diff
@@ -341,16 +341,32 @@ export function patchRecordMethods(modelName, recordMethods) {
 }
 
 /**
- * @param {Object} definition The JSON definition of the model to register.
- * @param {Object} [definition.fields]
- * @param {string[]} definition.identifyingFields
- * @param {Object} [definition.lifecycleHooks]
- * @param {Object} [definition.modelGetters] Deprecated; use fields instead.
- * @param {Object} [definition.modelMethods]
- * @param {string} definition.name
- * @param {OnChanges[]} [definition.onChanges]
- * @param {Object} [definition.recordGetters] Deprecated; use fields instead.
- * @param {Object} [definition.recordMethods]
+ * @typedef {Object} LifecycleHooks
```

Comment:
`typedef` is generic across all Odoo or specific to this file?
If generic it should be more specific, maybe at least `RecordLifecycleHook`.
(and it should be singular no matter what)

Reply:
Actually for singular this is not really correct either, we always try to avoid plural names, but just singular is not correct in this case. Should be `LifecycleHooksDefinition` maybe?

Reply:
This is for this file, but could be imported some how.
Anyway, `LifecycleHooksDefinition` look life a better name. But Definition is redundant with `typedef`.
I don't mind the singular here

Reply:
Definition is not related to the typedef, definition is because it is the object contains the definition of lifecycle hooks.

Reply:
Alright, I don't care much as long as the typedef exist, it's ok for me.